File: cpp/practice/Lab01/CircleV1/Circle.cpp
#include "Circle.hpp"

Circle::Circle() {
    _radius = 0;
    _center = Point();
}

Circle::Circle(double r, Point c) {
    _radius = r;
    _center = c;
}

Circle::~Circle() {
    // nothing to do
}

double Circle::getRadius() const {
    return _radius;
}

Point Circle::getCenter() const {
    return _center;
}

void Circle::setRadius(double r) {
    _radius = r;
}

void Circle::setCenter(Point c) {
    _center = c;
}

double Circle::getArea() const {
    return PI * _radius * _radius;
}

double Circle::getPerimeter() const {
    return 2 * PI * _radius;
}

ostream& operator<<(ostream& os, const Circle c) {
    os << "Circle: radius = " << c._radius << ", center = " << c._center;
    return os;
}

istream& operator>>(istream& is, Circle& c) {
    cout << "Enter radius: ";
    is >> c._radius;
    cout << "Enter center: ";
    is >> c._center;
    return is;
}

File: java/solutions/Lab7/Circle.java
public class Circle implements GeometricObject {
    protected double radius;

    public Circle() {
        radius = 0 ;
    }

    public Circle(double radius) {
        this.radius = radius;
    }

    @Override
    public double getArea() {
        return radius * radius * 3.14;
    }

    @Override
    public double getPerimeter() {
        return 2 * 3.14 * radius;
    }

}